fairscale/nn/misc/checkpoint_activations.py

```diff
@@ -32,9 +32,25 @@ def checkpoint_wrapper(module: nn.Module, offload_to_cpu: bool = False) -> nn.Mo
 
     Args:
         module (nn.Module):
-            module to wrap
+            The module to be wrapped
         offload_to_cpu (Optional, bool):
-            whether to offload activations to CPU
+            Whether to offload activations to CPU. This is non-trivial
+            to use. Enable it doesn't always cause lower peak GPU memory.
+            It can cause more GPU memory being used if used incorrectly.
+
+                - layer with the model input doesn't benefit from offloading
+                  since the input tensor is hold in GPU anyway, offloading
+                  can cause more memory being used
+                - last layer of the model doesn't benefit since immediately
+                  we load the tensor back from CPU in the backward
+                - long model with a lot of checkpoint blocks and small
+                  inner (recomputed) activations is more likely to benefit
+                  from this. But bigger checkpoint blocks might be better
+                  in that case.
```

Comment:
Let's chat more about this. Here's a good repro I shared with Sam that illustrates when/where offload_to_cpu is useful: https://gist.github.com/myleott/7d2ea5a8d27cb6443948e9fa173faf8f#gistcomment-3642158